tests/L0/run_fused_layer_norm/test_fused_layer_norm.py

```diff
@@ -8,291 +8,282 @@
-        torch.testing.assert_allclose(native_x.grad, fused_x.grad, **tols)
-
-    def test_autocast(self):
-        for (dtype, elementwise_affine) in itertools.product(autocast_dtypes, (True, False)):
-            self._run_test(dtype, elementwise_affine)
+        tols = {'rtol': None, 'atol': None} if dtype == torch.half else bf16_bwd_thresholds
+        torch.testing.assert_close(native_x.grad, fused_x.grad, **tols, check_dtype=False)
 
-class TestAutocastFusedRMSNorm(unittest.TestCase):
-    bf16_fwd_thresholds = dict(rtol=1.6e-2, atol=3e-4)
-    bf16_bwd_thresholds = dict(rtol=1.6e-2, atol=3e-3)
 
-    def setUp(self):
-        self.batch_size = 16
-        self.normalized_shape = [32, 16]
-
-    def _run_test(self, dtype, elementwise_affine):
-        native, fused = _prep_rms_layers(self.normalized_shape, elementwise_affine, dtype)
-        native_x, fused_x = _prep_inputs(self.batch_size, self.normalized_shape, dtype)
+    @common_utils.parametrize(
+        "dtype, elementwise_affine",
+        list(product(autocast_dtypes, (True, False)))
+    )
+    def test_autocast_fused_rms_norm(self, dtype, elementwise_affine):
+        bf16_fwd_thresholds = dict(rtol=1.6e-2, atol=3e-4)
+        bf16_bwd_thresholds = dict(rtol=1.6e-2, atol=3e-3)
+        batch_size = 16
+        normalized_shape = [32, 16]
+        native, fused = _prep_rms_layers(normalized_shape, elementwise_affine, dtype)
+        native_x, fused_x = _prep_inputs(batch_size, normalized_shape, dtype)
 
         expected = native(native_x.cpu())
         with torch.cuda.amp.autocast(dtype=dtype):
             actual = fused(fused_x)
-        tols = {'rtol': None, 'atol': None} if dtype == torch.half else TestAutocastFusedRMSNorm.bf16_fwd_thresholds
-        torch.testing.assert_allclose(actual, expected.detach().clone().cuda(), **tols)
+        tols = {'rtol': None, 'atol': None} if dtype == torch.half else bf16_fwd_thresholds
+        torch.testing.assert_close(actual, expected.detach().clone().cuda(), **tols, check_dtype=False)
 
         g_native = torch.rand_like(expected)
         with torch.no_grad():
             g_fused = g_native.detach().clone().cuda()
         expected.backward(g_native)
         actual.backward(g_fused)
 
-        tols = {'rtol': None, 'atol': None} if dtype == torch.half else TestAutocastFusedRMSNorm.bf16_bwd_thresholds
-        torch.testing.assert_allclose(native_x.grad.cuda(), fused_x.grad, **tols)
+        tols = {'rtol': 1e-3, 'atol': 1e-3} if dtype == torch.half else bf16_bwd_thresholds
+        torch.testing.assert_close(native_x.grad.cuda(), fused_x.grad, **tols, check_dtype=False)
 
-    def test_autocast(self):
-        for (dtype, elementwise_affine) in itertools.product(autocast_dtypes, (True, False)):
-            self._run_test(dtype, elementwise_affine)
 
+instantiate_device_type_tests(TestFusedLayerNorm, globals(), only_for=("cuda",))
 
 if __name__ == "__main__":
-    unittest.main()
+    common_utils.run_tests()
```

Comment:
could you add a new line at the end of file?
if you're using vscode, `"files.insertFinalNewline": true` to the setting file would do

Reply:
Fixed